src/Integrations/ActiveCampaign/SettingsActiveCampaign.php

```diff
@@ -0,0 +1,383 @@
+<?php
+
+/**
+ * ActiveCampaign Settings class.
+ *
+ * @package EightshiftForms\Integrations\ActiveCampaign
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Integrations\ActiveCampaign;
+
+use EightshiftFormsVendor\EightshiftLibs\Helpers\Components;
+use EightshiftForms\Helpers\Helper;
+use EightshiftForms\Hooks\Filters;
+use EightshiftForms\Settings\SettingsHelper;
+use EightshiftForms\Hooks\Variables;
+use EightshiftForms\Integrations\ActiveCampaign\ActiveCampaignClientInterface;
+use EightshiftForms\Integrations\MapperInterface;
+use EightshiftForms\Settings\GlobalSettings\SettingsGlobalDataInterface;
+use EightshiftForms\Settings\Settings\SettingsDataInterface;
+use EightshiftFormsVendor\EightshiftLibs\Services\ServiceInterface;
+
+/**
+ * SettingsActiveCampaign class.
+ */
+class SettingsActiveCampaign implements SettingsDataInterface, SettingsGlobalDataInterface, ServiceInterface
+{
+	/**
+	 * Use general helper trait.
+	 */
+	use SettingsHelper;
+
+	/**
+	 * Filter settings sidebar key.
+	 */
+	public const FILTER_SETTINGS_SIDEBAR_NAME = 'es_forms_settings_sidebar_active_campaign';
+
+	/**
+	 * Filter settings key.
+	 */
+	public const FILTER_SETTINGS_NAME = 'es_forms_settings_active_campaign';
+
+	/**
+	 * Filter global settings key.
+	 */
+	public const FILTER_SETTINGS_GLOBAL_NAME = 'es_forms_settings_global_active_campaign';
+
+	/**
+	 * Filter settings is Valid key.
+	 */
+	public const FILTER_SETTINGS_IS_VALID_NAME = 'es_forms_settings_is_valid_active_campaign';
+
+	/**
+	 * Settings key.
+	 */
+	public const SETTINGS_TYPE_KEY = 'active-campaign';
+
+	/**
+	 * ActiveCampaign Use key.
+	 */
+	public const SETTINGS_ACTIVE_CAMPAIGN_USE_KEY = 'active-campaign-use';
+
+	/**
+	 * API Key.
+	 */
+	public const SETTINGS_ACTIVE_CAMPAIGN_API_KEY_KEY = 'active-campaign-api-key';
+
+	/**
+	 * API Url.
+	 */
+	public const SETTINGS_ACTIVE_CAMPAIGN_API_URL_KEY = 'active-campaign-api-url';
+
+	/**
+	 * List ID Key.
+	 */
+	public const SETTINGS_ACTIVE_CAMPAIGN_LIST_KEY = 'active-campaign-list';
+
+	/**
+	 * Integration fields Key.
+	 */
+	public const SETTINGS_ACTIVE_CAMPAIGN_INTEGRATION_FIELDS_KEY = 'active-campaign-integration-fields';
+
+	/**
+	 * Instance variable for ActiveCampaign data.
+	 *
+	 * @var ActiveCampaignClientInterface
+	 */
+	protected $activeCampaignClient;
+
+	/**
+	 * Instance variable for ActiveCampaign form data.
+	 *
+	 * @var MapperInterface
+	 */
+	protected $activeCampaign;
```

Comment:
Why are these `protected` and not `private`?

Reply:
agreed fixed